src/BaseLoginForm.jsx

```diff
@@ -36,23 +36,28 @@ class BaseLoginForm extends Component {
         this.onTokenChange = this.onTokenChange.bind(this);
     }
     componentDidMount() {
-        const {auth} = this.props;
+        const {auth, onChange} = this.props;
         auth.subscribeTokenChange(this.onTokenChange);
-        auth.getToken().then(this.props.onChange('token'));
 
-
-        // Force update token after subscription
+        onChange('requestState')(FetchingState());
         auth
-            .refreshToken()
+            .getToken()
+            .then((token) => {
+                this.onTokenChange(token);
+                return auth.refreshToken();
+            })
+            .then(() => {
+                onChange({
+                    isTokenValid: true,
+                    requestState: SuccessState()
+                });
+            })
             .catch(err => {
                 if(process.env.NODE_ENV === 'development') {
```

Comment:
I think you can turf this line now. The problem is solved at it's root, we can still see the failed request in the console.